server/routes.go

```diff
@@ -844,6 +844,9 @@ func GetModelInfo(req api.ShowRequest) (*api.ShowResponse, error) {
 
 	for k, v := range req.Options {
 		if _, ok := req.Options[k]; ok {
+			if m.Options == nil {
+				m.Options = make(map[string]any)
+			}
 			m.Options[k] = v
 		}
 	}
```

Comment:
This options building code looks kinda awkward, I think this would be simpler:

```suggestion
if len(req.Options) > 0 {
    if m.Options == nil {
        m.Options = make(map[string]any)
    }
    for k, v := range req.Options {
        m.Options[k] = v
    }
}
```

I dont see a reason for the `ok` assertion and checking m.Options being nil on each iteration